cli/cartridge/extra/003_fix_work_paths.patch

```diff
@@ -0,0 +1,130 @@
+diff --git a/cli/admin/common.go b/cli/admin/common.go
+index dcea75c..e3a6dd7 100644
+--- a/cli/admin/common.go
++++ b/cli/admin/common.go
+@@ -2,7 +2,6 @@ package admin
+ 
+ import (
+ 	"fmt"
+-	"io/ioutil"
+ 	"os"
+ 	"path/filepath"
+ 	"sort"
+@@ -140,31 +139,30 @@ func getInstanceSocketPaths(ctx *context.Ctx) ([]string, error) {
+ 		return nil, fmt.Errorf("%s is not a directory", ctx.Running.RunDir)
+ 	}
+ 
+-	runFiles, err := ioutil.ReadDir(ctx.Running.RunDir)
+-	if err != nil {
+-		return nil, fmt.Errorf("Failed to list the run directory: %s", err)
+-	}
+-
+-	if len(runFiles) == 0 {
+-		return nil, fmt.Errorf("Run directory %s is empty", ctx.Running.RunDir)
+-	}
+-
+ 	instanceSocketPaths := []string{}
+ 
+-	appInstanceSocketPrefix := fmt.Sprintf("%s.", ctx.Project.Name)
+-	controlSocketSuffix := ".control"
+-	for _, runFile := range runFiles {
+-		runFileName := runFile.Name()
+-		if !strings.HasSuffix(runFileName, controlSocketSuffix) {
+-			continue
+-		}
+-
+-		if !strings.HasPrefix(runFileName, appInstanceSocketPrefix) {
+-			continue
+-		}
+-
+-		instanceSocketPath := filepath.Join(ctx.Running.RunDir, runFileName)
+-		instanceSocketPaths = append(instanceSocketPaths, instanceSocketPath)
++	err := filepath.Walk(ctx.Running.RunDir,
++		func(path string, info os.FileInfo, err error) error {
++			if err != nil {
++				return err
++			}
++			controlSocketSuffix := ".control"
++
++			if info.Mode()&os.ModeSocket == os.ModeSocket {
++				file := filepath.Base(path)
++				if !strings.HasSuffix(file, controlSocketSuffix) {
++					return nil
++				}
++
++				parts := strings.Split(path, "/")
++				instanceSocketPath := filepath.Join(ctx.Running.RunDir, parts[len(parts)-2])
```

Comment:
I need to cut only parent folder here, filepath.Dir cannot do the same.

Reply:
But using filepath api is safer. For instance, it will process this path correctly:
```go
"/a/b/c////socket.aaa"
filepath.Base(filepath.Dir(filename)) // will give "c"
parts[len(parts)-2] // will give an empty string
```

Reply:
fixed